src/url.cpp

```diff
@@ -782,12 +782,24 @@ bool url::set_port(const std::string_view input) {
     port = std::nullopt;
     return true;
   }
+
+  if (!ada::unicode::is_ascii_digit(trimmed.front())) {
+    return false;
+  }
+
+  auto non_digit_pos = std::find_if_not(trimmed.begin(), trimmed.end(),
+                                        ada::unicode::is_ascii_digit);
+  if (non_digit_pos != trimmed.end()) {
+    trimmed.erase(non_digit_pos, trimmed.end());
+  }
+
   // Input should not start with control characters.
   if (ada::unicode::is_c0_control_or_space(trimmed.front())) {
     return false;
   }
-  // Input should contain at least one ascii digit.
-  if (input.find_first_of("0123456789") == std::string_view::npos) {
+
+  // Input should contain ascii digits.
+  if (!std::ranges::all_of(trimmed, ada::unicode::is_ascii_digit)) {
     return false;
   }
```

Comment:
Since we are checking if the first character is not a digit or not, we don't need this unnnessary check:

```suggestion
```